contracts/stellar-axelar-gateway/src/contract.rs

```diff
@@ -161,21 +163,21 @@ impl AxelarGatewayInterface for AxelarGateway {
         ensure!(!messages.is_empty(), ContractError::EmptyMessages);
 
         for message in messages.into_iter() {
-            let message_approval_key = MessageApprovalKey {
-                source_chain: message.source_chain.clone(),
-                message_id: message.message_id.clone(),
-            };
-
             // Prevent replay if message is already approved/executed
-            let message_approval = storage::try_message_approval(env, message_approval_key.clone())
-                .unwrap_or(MessageApprovalValue::NotApproved);
-            if message_approval != MessageApprovalValue::NotApproved {
+            if storage::try_message_approval(
+                env,
+                message.source_chain.clone(),
+                message.message_id.clone(),
+            )
+            .is_some()
```

Comment:
```suggestion
            if storage::has_message_approval(
                env,
                message.source_chain.clone(),
                message.message_id.clone(),
            )
```
maybe consider adding a `has_*` storage method